plan/ctas_plan_test.go

```diff
@@ -0,0 +1,168 @@
+				&types.DecimalType{Precision: 10, Scale: 2, Nullability: types.NullabilityNullable},
+				&types.StringType{Nullability: types.NullabilityNullable},
+			},
+		}}
+)
+
+// makeProjectionMaskExpr generates a MaskExpression to project or reorder columns by the given IDs.
+func makeProjectionMaskExpr(columnIds []int) *expr.MaskExpression {
+	structItems := make([]*substraitproto.Expression_MaskExpression_StructItem, len(columnIds))
+
+	for index, columnId := range columnIds {
+		structItems[index] = &substraitproto.Expression_MaskExpression_StructItem{
+			Field: int32(columnId),
+		}
+	}
+
+	return expr.MaskExpressionFromProto(
+		&substraitproto.Expression_MaskExpression{
+			Select: &substraitproto.Expression_MaskExpression_StructSelect{
+				StructItems: structItems,
+			},
+			MaintainSingularStruct: true,
+		},
+	)
+}
+
+// makeNamedTableReadRel creates a named table read relation with the selected column IDs.
+func makeNamedTableReadRel(b plan.Builder, tableNames []string, tableSchema types.NamedStruct, columnIds []int) plan.Rel {
+	namedTableReadRel := b.NamedScan(tableNames, tableSchema)
+	namedTableReadRel.SetProjection(makeProjectionMaskExpr(columnIds))
+	return namedTableReadRel
+}
+
+// makeConditionExprForLike constructs a LIKE condition expression for the specified column and value.
+func makeConditionExprForLike(t *testing.T, b plan.Builder, scan plan.Rel, colId int, valueLiteral expr.Literal) expr.Expression {
+	id := extensions.ID{
+		"https://github.com/substrait-io/substrait/blob/main/extensions/functions_string.yaml",
+		"contains:str_str",
+	}
+	b.GetFunctionRef(id.URI, id.Name)
+	colIdRef, err := b.RootFieldRef(scan, int32(colId))
+	require.NoError(t, err)
+	scalarExpr, err := b.ScalarFn(id.URI, id.Name, nil, colIdRef, valueLiteral)
+	require.NoError(t, err)
+	return scalarExpr
+}
+
+func makeFilterRel(t *testing.T, b plan.Builder, input plan.Rel, condition expr.Expression) plan.Rel {
+	filterRel, err := b.Filter(input, condition)
+	require.NoError(t, err)
+	return filterRel
+}
+
+func makeProjectRel(t *testing.T, b plan.Builder, input plan.Rel, columnIds []int) plan.Rel {
+	refs := make([]expr.Expression, len(columnIds))
+	for i, c := range columnIds {
+		ref, err := b.RootFieldRef(input, int32(c))
+		require.NoError(t, err)
+		refs[i] = ref
+	}
+	project, err := b.Project(input, refs...)
+	require.NoError(t, err)
+	return project
+}
+
+// getProjectionForTest builds the SELECT portion of the CTAS statement based on the provided test name.
+func getProjectionForTest(t *testing.T, b plan.Builder, testName string) plan.Rel {
+	// Select name, salary from employees
+	if strings.EqualFold(testName, "ctas_basic") {
+		namedScanRel := makeNamedTableReadRel(b, []string{"employees"}, employeeSchema, []int{1, 3})
+		return makeProjectRel(t, b, namedScanRel, []int{0, 1})
+	}
+
+	// Select * from employees where role LIKE 'Engineer'
+	if strings.EqualFold(testName, "ctas_with_filter") {
```

Comment:
Optional:  It's almost worth having separate functions for each generated schema so we could pass a function reference as part of the test case.

Reply:
Thanks @EpsilonPrime . I like this idea.